lib/msal-core/src/AuthError.ts

```diff
@@ -0,0 +1,240 @@
+            ErrorMessage.noCallback.desc);
+    }
+
+    static createErrorInCallbackFunction(errorDesc: string): ClientAuthError {
+        return new ConfigurationError(ErrorMessage.callbackError.code,
+            ErrorMessage.callbackError.desc + errorDesc);
+    }
+
+    // TODO: Add these to Error.ts in case if they are uncommented/not deleted later
+    // static createCallbackParametersError(numArgs: number, tokenType: string, userState: string) : ClientAuthError {
+    //     return new ClientConfigurationAuthError("Error occurred in callback - incorrect number of arguments, expected 2, got " + numArgs + ".",
+    //         tokenType,
+    //         userState);
+    // }
+
+    // static createSuccessCallbackParametersError(numArgs: number, tokenType: string, userState: string) : ClientAuthError {
+    //     return new ConfigurationAuthError("Error occurred in callback for successful token response - incorrect number of arguments, expected 1, got " + numArgs + ".", tokenType, userState);
+    // }
+
+    // static createErrorCallbackParametersError(numArgs: number, tokenType: string, userState: string) : ClientAuthError {
+    //     return new ConfigurationAuthError("Error occurred in callback for error response - incorrect number of arguments, expected 1, got " + numArgs + ".", tokenType, userState);
+    // }
+
+    static createEmptyScopesArrayError(scopesValue: string): ConfigurationError {
+        return new ConfigurationError(ErrorMessage.emptyScopes.code,
+            ErrorMessage.emptyScopes.desc + ". Given value: " + scopesValue);
+    }
+
+    static createScopesNonArrayError(scopesValue: string): ConfigurationError {
+        return new ConfigurationError(ErrorMessage.nonArrayScopes.code,
+            ErrorMessage.nonArrayScopes.desc + " Given value: " + scopesValue);
+    }
+
+    static createClientIdSingleScopeError(scopesValue: string): ConfigurationError {
+        return new ConfigurationError(ErrorMessage.clientScope.code,
+            ErrorMessage.clientScope.desc + " Given value: " + scopesValue);
+    }
+}
+
+/**
+ * @hidden
+ *
+ * Error thrown when there is an error with the server code, for example, unavailability.
+ */
+export class ServerError extends AuthError {
+    constructor(error: string, errorDesc: string) {
+        super(error, errorDesc);
+        this.name = "ServerError";
+
+        Object.setPrototypeOf(this, ServerError.prototype);
+    }
+
+    static createServerUnavailableError(): ServerError {
+        return new ServerError(ErrorMessage.serverUnavailable.code,
+            ErrorMessage.serverUnavailable.desc);
+    }
+
+    static createUnknownServerError(errorDesc: string): ServerError {
+        return new ServerError(ErrorMessage.unknownServerError.code,
+            errorDesc);
+    }
+}
+
+/**
+ * @hidden
+ *
+ * Error thrown when the user is required to perform an interactive token request.
+ */
+export class InteractionRequiredAuthError extends ServerError {
```

Comment:
What does extending server error get us ? There is no chain right?

Reply:
We do have a prototype chain. That is the entire point of setPrototypeOf().

Reply:
I mean a chain in terms of telling errors parents from eachother, but you answered my question above for the value. Marking resolved